src/components/LoginPage/LoginPageMain.jsx

```diff
@@ -0,0 +1,138 @@
+import React, { useState } from "react";
+import "../LoginPage/LoginPage.css";
+import email_image from "../assets/email.png";
+import person_image from "../assets/person.png";
+import password_image from "../assets/password.png";
+
+const LoginPageMain = () => {
+  const [action, setAction] = useState("Login");
+  const [email, setEmail] = useState("");
+  const [password, setPassword] = useState("");
+  const [name, setName] = useState("");
+  const [loginError, setLoginError] = useState("");
+  const [isLoggedIn, setIsLoggedIn] = useState(false);
+
+  const [users, setUsers] = useState([]);
+
+  const handleLogin = () => {
+    const user = users.find(
+      (user) => user.email === email && user.password === password
+    );
+
+    if (user) {
+      setIsLoggedIn(true);
+      setLoginError("");
+    } else {
+      setLoginError("Invalid email or password");
+    }
+  };
+
+  const handleSignUp = () => {
+    if (users.find((user) => user.email === email)) {
+      setLoginError("User with this email already exists.");
+      return;
+    }
+
+    if (name && email && password) {
+      setUsers([...users, { name, email, password }]);
+      alert("Sign-up successful! You can now log in.");
+      setAction("Login");
+      setLoginError("");
+      setName("");
+      setEmail("");
+      setPassword("");
+    } else {
+      setLoginError("All fields are required for sign-up.");
+    }
+  };
+
+  const handleLogout = () => {
+    setIsLoggedIn(false);
+    setEmail("");
+    setPassword("");
+    setName("");
+    setAction("Login");
+  };
+
+  return (
```

Comment:
This is a very long tree of components. There's also a lot of switch cases (switch what we show based on a variable's value).

I think it would be better to split login and signup into two different components at two different routes `/login` and `/signup`. We can navigate the user to the other route if they click the button to go to the other one.

Eventually, once we hook up this code to the API, it's going to be really complicated and annoying to have both login and signup as part of one component.